src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void createNewArray() {
+        int newSize = array.length + array.length / 2;
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
+        array = newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            createNewArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
 
+        if (index < 0 && index > size) {
```

Comment:
The condition should use `||` (logical OR) instead of `&&` (logical AND), as you want to throw an exception if the index is either less than 0 or greater than size. This is a bug in the index validation logic.